src/linear_algebra/contraction.lean

```diff
@@ -110,3 +128,127 @@ dual_tensor_hom_equiv_of_basis (module.free.choose_basis R M)
 end comm_ring
 
 end contraction
+
+section hom_tensor_hom
+
+open_locale tensor_product
+
+open module tensor_product linear_map
+
+section comm_semiring
+
+variables [comm_semiring R] [module R M] [module R N] [module R P] [module R Q]
+
+/-- The canonical linear map from `P ⊗[R] (M →ₗ[R] Q)` to `(M →ₗ[R] P ⊗[R] Q)` -/
+def ltensor_hom_to_hom_ltensor : P ⊗[R] (M →ₗ[R] Q) →ₗ[R] (M →ₗ[R] P ⊗[R] Q) :=
```

Comment:
The `to` version of these two defs can probably go into tensor_product.lean since they don't mention duals. The `equiv` version is constructed using duals so they need to remain.